src/main/java/org/gradle/profiler/BuildUnderTestInvoker.java

```diff
@@ -80,10 +80,10 @@ public GradleBuildInvocationResult run(BuildContext buildContext, BuildStep buil
             return new GradleBuildInvocationResult(
                 buildContext,
                 buildActionResult.getExecutionTime(),
+                buildActionResult.getGradleToolingAgentExecutionTime(),
+                buildActionResult.getIdeExecutionTime(),
```

Comment:
Actually, I think these should be `null`s here (and hence we should go for `Optional` in `BuildActionResult`). Otherwise even non-IDE benchmarks will have these measurements, no?

Reply:
Makes sense for the consistency, fixed.  Note that for non-IDE benchmarks we don't read these values so these values are not there.